File: XMLCheck/Handler/CheckRulesXML11/Xml11RuleRunner.cs
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML11
{
    public static class Xml11RuleRunner
    {
        private static readonly List<IXml11Rule> _rules = new()
        {

        };
        public static List<ErrorDetails> Run(Xml11Model? model,HoSoContext context)
        {
            var errors = new List<ErrorDetails>();

            foreach (var rule in _rules)
            {
                var err = rule.Check(model,context);
                if (err != null)
                    errors.Add(err);
            }

            return errors;
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml11/xml11RuleRunner.ts
import { Xml11Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';
import { IXml11Rule } from './xml11RuleBase';

export class Xml11RuleRunner {
  private static readonly rules: IXml11Rule[] = [];

  public static run(model: Xml11Model, context: HoSoContext): ErrorDetails[] {
    const errors: ErrorDetails[] = [];
    if (!model) return errors;

    for (const rule of this.rules) {
      const err = rule.check(model, context);
      if (err) {
        errors.push(err);
      }
    }

    return errors;
  }
}
